build.sh

```diff
@@ -1,8 +1,75 @@
-git clone https://github.com/ufcg-lsd/asperathos-controller ./controller/asperathos-controller
-docker build controller/ -t asperathos_controller
-git clone https://github.com/ufcg-lsd/asperathos-monitor ./monitor/asperathos-monitor
-docker build monitor/ -t asperathos_monitor
-git clone https://github.com/ufcg-lsd/asperathos-manager ./manager/asperathos-manager
-docker build manager/ -t asperathos_manager
-git clone https://github.com/ufcg-lsd/asperathos-visualizer ./visualizer/asperathos-visualizer
-docker build visualizer/ -t asperathos_visualizer
+#!/bin/bash
+
+helpFunction()
+{
+   echo ""
+   echo "Usage: $0 -b BRANCH [-r] [-l] [-h]"
+   echo -e "\t-b BRANCH      Select Branch        The branch you want to clone from (the same branch will be used for all projects). If you use -b with a branch different from the current one the files will be deleted and the new branch will be downloaded"
+   echo -e "\t-r             Reset                If you want to reset the current asperathos files"
+   echo -e "\t-l             Local Rebuild        If you want to rebuild using the local files. If you use this flag the other's will be ignored."
+   exit 1
+}
+
+DIRECTORY="asperathos-manager"
+
+DEFAULT_BRANCH="stable"
+
+if [ -d "$DIRECTORY" ]; then
+    cd asperathos-manager
+    CURRENT_BRANCH=$(git branch | grep \* | cut -d ' ' -f2)
+    cd ../..
+fi
+
+
+while getopts b:rlh option
+do
+    case "${option}"
+    in
+        b) BRANCH="$OPTARG" ;;
+        r) RESET="true" ;;
+        l) LOCAL="true" ;;
+        h) HELP="true" ;;
+        ?) exit 1
+    esac
+done
+
+
+if [ "$HELP" ]
+then
+    helpFunction
+fi
+
+
+if [ -z "$BRANCH" ]
+then
+    BRANCH=$DEFAULT_BRANCH
+fi
+
+
+if [[ -z "$LOCAL"  &&  "$RESET" ]] || [[ -z "$LOCAL"  && "$CURRENT_BRANCH" != "$BRANCH" ]]
+then
+    echo "Deleting existing files"
+    rm -rf asperathos-controller/
+    rm -rf asperathos-visualizer/
+    rm -rf asperathos-manager/
+    rm -rf asperathos-monitor/
+fi
+
+
+
+
+
+if [ -z "$LOCAL" ]
+then
+    echo "Cloning the branch ${BRANCH}"
```

Comment:
Just a personal preference, when something is currently happening then I like to finish log sentences with "..." so it gives the feeling it is ongoing.